communicator/shared/shared.go

```diff
@@ -0,0 +1,26 @@
+package shared
+
+import (
+	"fmt"
+	"net"
+)
+
+// HostFormatter helps out with host formatting.
+// For instance, the host parameter differs when connecting to an IPv4 address vs. an IPv6 address.
+type HostFormatter interface {
+	Format(string) string
+}
+
+// HostFormatterImpl implements the HostFormatter interface.
+type HostFormatterImpl struct{}
+
+// Format formats the host/IP correctly, so we don't provide IPv6 address in an IPv4 format during node communication.
+func (h *HostFormatterImpl) Format(host string) string {
```

Comment:
is there a reason you're implementing this with a method receiver instead of just a function? Are you actually using this interface with any other types?